core/indexer/indexer.go

```diff
@@ -288,6 +294,28 @@ func (i *indexer) getObjectInfo(ctx context.Context, id string) (info smartblock
 	return
 }
 
+func (i *indexer) setFileOrigin(hash string, origin int) {
+	fileOrigin, err := i.fileStore.GetFileOrigin(hash)
```

Comment:
This code should be inside `Details(ctx context.Context) (*types.Struct, error)` in
core/files/image.go
core/files/file.go

Reply:
Done